docker-compose.override.yml

```diff
@@ -9,6 +9,8 @@ services:
     environment:
       - PROJ_LIB=/usr/share/proj  # This is to fix an odd issue with PyProj
       - prometheus_multiproc_dir=/tmp # Enable prometheus metrics
+      - DATACUBE_CONFIG_PATH=/root/.datacube_integration.conf # Enable test config
+      - DB_DATABASE=opendatacube_test
```

Comment:
This means it'll always use the test DB, right?

When you run the integration tests, it uses the `integration` config file, I believe. It's built in...

I don't think you need to make these two changes.

Reply:
no, this is to actually make this file useable for manual testing

Reply:
I don't believe so, @pindge 

Datacube Core has a special case for config for running integration tests: https://github.com/opendatacube/datacube-core/blob/c2c25d3df437218223b900c1e6deb6f3dce374e4/README.rst#developer-setup

I don't think we necessarily want to use the `opendatacube_test` database for doing development work, which is what setting `DB_DATABASE` here will do.

Reply:
@jeremyh  can confirm here 

But what you'll find is if you remove the volume mount and these two environment variables, the app will work fine. But integration tests will fail.

If you leave it the way I initially set it up, it works fine as a DEV install, and integration tests also work.

Reply:
This file name is misleading, if its only to be used for integration time, it shouldn't be called `.override` should be `docker-compose.integration.yaml`

Reply:
It's not meant only for integration time...

The override is the default, it's used for DEV.

In the Makefile, there's a `-prod` set of commands, that work without the override. That means you can test a build that doesn't have the code mounted, which is convenient for testing some things, like whether it gets installed right, and whether the version number is captured properly.

It is a bit complex, and could do with documenting! But it works, and I'm saying we don't need to change these aspects of it.